__tests__/integration.test.js

```diff
@@ -153,6 +153,78 @@ describe('POST/api/articles/:article_id/comments', () => {
     });
 });
 
+describe('PATCH: api/articles/:article_id', () => {
+    test('200: identifies article id and adds to an articles votes value when given integer', () => {
+        const articleVotes = {
+            inc_votes: 8
+        };
+        return request(app)
+        .patch('/api/articles/3')
+        .send(articleVotes)
+        .expect(201)
+        .then(({ body }) => {
+            expect(body.votes).toBe(8)
+            expect(body).toHaveProperty("author");
+            expect(body).toHaveProperty("title");
+            expect(body).toHaveProperty("article_id");
+            expect(body).toHaveProperty("topic");
+            expect(body).toHaveProperty("created_at");
+            expect(body).toHaveProperty("votes");
+            expect(body).toHaveProperty("article_img_url");
+        });
+    });
+    test('200: identifies article id and subtracts from an articles votes value when given a negative number', () => {
+        const articleVotes = {
+            inc_votes: -15
+        };
+        return request(app)
+        .patch('/api/articles/3')
+        .send(articleVotes)
+        .expect(201)
+        .then(({ body }) => {
+            expect(body.votes).toBe(-15)
+            expect(body).toHaveProperty("author");
+            expect(body).toHaveProperty("title");
+            expect(body).toHaveProperty("article_id");
+            expect(body).toHaveProperty("topic");
+            expect(body).toHaveProperty("created_at");
+            expect(body).toHaveProperty("votes");
+            expect(body).toHaveProperty("article_img_url");
+        });
+    });
+    test('200: identifies article id and adds to articles votes when vote count is not initially 0', () => {
+        const articleVotes = {
+            inc_votes: 27
+        };
+        return request(app)
+        .patch('/api/articles/1')
+        .send(articleVotes)
+        .expect(201)
+        .then(({ body }) => {
+            expect(body.votes).toBe(127)
+            expect(body).toHaveProperty("author");
+            expect(body).toHaveProperty("title");
+            expect(body).toHaveProperty("article_id");
+            expect(body).toHaveProperty("topic");
+            expect(body).toHaveProperty("created_at");
+            expect(body).toHaveProperty("votes");
+            expect(body).toHaveProperty("article_img_url");
+        });
+    });
+    test('400: sends appropriate error message when passed an invalid key type', () => {
+        const badArticleVotes = {
+            inc_votes: "hello"
+        };
+        return request(app)
+        .patch('/api/articles/1')
+        .send(badArticleVotes)
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request")
+        })
+    })
+});
+
```

Comment:
Dont forget to include error handling tests for when the article id is invalid i.e. `/api/articles/banana` and when the article id is valid but does not exist i.e. `/api/articles/9999`